src/input/pointer.js

```diff
@@ -314,8 +383,8 @@
     function updateCoordFromEvent(event) {
         var local;
 
-        // reset the touch array cache
-        changedTouches.length = 0;
+        // touch array cache
+        var changedTouches = [];
```

Comment:
You're right. In a previous implementation, when a move event needed to be re-dispatched as a enter/leave event, the global changedTouches caused many problems. I think I can switch it again to the global one, it should work with the current implementation.